documentation/reference/charm.md

```diff
@@ -44,3 +45,19 @@ flowchart TD
   match_cfg --> render_cfg[Render config &\nreload pgbouncer]
   render_cfg --> rtn2([return])
 ```
+
+### Status Changed Hook
+
+```mermaid
+flowchart TD
+  hook_fired([update-status Hook]) --> is_backend_ready{Is backend\ndatabase ready?}
+  is_backend_ready -- no --> set_blocked[Set Blocked\nstatus]
+  is_backend_ready -- yes --> is_running{Is pgbouncer\nrunning?}
+  is_running -- yes --> set_active[Set Active\n Status]
+  is_running -- no --> set_waiting[Set Waiting\nStatus]
+  set_blocked --> update_leader[Update leader in\nPeer Relation\nDatabag.]
```

Comment:
It seems that the blocked status is overwritten by one of the other two status on https://github.com/canonical/pgbouncer-k8s-operator/blob/8e58877ea65ea08f143f092b72bf3200aba5fe2d/src/charm.py#L213.